src/configuration.jsx

```diff
@@ -314,9 +314,14 @@ class Configuration extends React.PureComponent {
 	};
 
 	handleSpecialDateAdd = ( newSpecialDate ) => {
-		const newSpecialDates = [ ...this.state.specialDates ];
-		newSpecialDates.push( wrapWithId( newSpecialDate ) );
-		this.setState( { specialDates: newSpecialDates } );
+		this.setState(
+			( prev ) => ( {
+				specialDates: [
+					...prev.specialDates,
+					wrapWithId( newSpecialDate ),
+				],
+			} ),
+		);
```

Comment:
Had to fix the `setState` to use the callback form since many updates are happening within a single render cycle now.